core/eolearn/core/eodata_io.py

```diff
@@ -106,7 +111,7 @@ def remove_redundant_files(filesystem, eopatch_features, filesystem_features, cu
 def load_eopatch(eopatch, filesystem, patch_location, features=..., lazy_loading=False):
     """A utility function used by `EOPatch.load` method."""
     features = list(walk_filesystem(filesystem, patch_location, features))
-    loading_data = [FeatureIO(ftype, path, filesystem) for ftype, _, path in features]
+    loading_data: Iterable[Any] = [create_feature_io(ftype, path, filesystem) for ftype, _, path in features]
```

Comment:
Couldn't a type here be `List[FeatureIO]` and wouldn't this be automatically implied?

Reply:
sadly no, because a few lines down we overwrite it with the values inside FeatureIOs (unless the load is lazy), so we cannot be more precise than `Any`. And since we do it via `map` it cant even be `List`.

Reply:
Well the problem is because this is later iterated over and fed into the EOPatch so they need to be the same variable. If you look at the whole block you'll see that it's a bit of a hassle. I guess we could do the loading in the middle of assigning values, but that was not part of this update, so I didn't refactor to that extent.

Reply:
I see now :facepalm: 
Ok, let's leave it like this.